amp/repository/aim/view/scripts/reportWizard/prefilters.js

```diff
@@ -8,384 +8,390 @@ var REPORT_URL = '/rest/data/report/';
+        widgetFilter.auxId = auxId; // used only for advanced search.
+        this.showFilterWidget();
+    } else if (widgetFilter.reportContextId === 'workspace_editor') {
+        var id = new URL(window.location).searchParams.get('tId');
+        if (id) {
+            this.loadSavedFilterData(id, false);
+        } else {
+            this.showFilterWidget();
+        }
+    } else if (widgetFilter.gotSavedFilters !== true) {
+        this.loadSavedFilterData(widgetFilter.reportContextId, true);
+    } else {
+        this.showFilterWidget();
+    }
 };
 
 Filters.prototype.loadSavedFilterData = function (id, isReport) {
-	var self = this;
-	var url = isReport ? REPORT_URL : TEAM_URL;
-	$.ajax({
-		type: 'GET',
-		url: url + id,
-		success: function (data) {
-			filters = isReport ? data.reportMetadata.reportSpec.filters : data[FILTER_OBJECT];
-			widgetFilter.deserialize({filters: filters}, {silent: true});
-			self.showFilterWidget();
-		}
-	});
-	widgetFilter.gotSavedFilters = true;
+    var self = this;
+    var url = isReport ? REPORT_URL : TEAM_URL;
+    $.ajax({
+        type: 'GET',
+        url: url + id,
+        success: function (data) {
+            filters = isReport ? data.reportMetadata.reportSpec.filters : data[FILTER_OBJECT];
+            widgetFilter.deserialize({filters: filters}, {silent: true});
+            self.showFilterWidget();
+        }
+    });
+    widgetFilter.gotSavedFilters = true;
 };
 
 Filters.prototype.showFilterWidget = function () {
-	widgetFilter.showFilters();
-	$('#filter-popup').show();
+    widgetFilter.showFilters();
+    $('#filter-popup').show();
 };
 
-Filters.prototype.showSettings	= function(reportContextId) {
-	initFormatPopup();
-	this.saveFilters	= new SaveFilters(this, true);
-	var element = document.getElementById("customFormat");
-	element.style.display = "inline";
-        this.settingsPanel.setHeader(this.settingsPanelName);
-	this.settingsPanel.setBody(element);
-	this.settingsPanel.center();
-	this.settingsPanel.show();
-	
-	YAHOO.util.Event.removeListener("applyFormatBtn", "click");
-	document.getElementById("applyFormatBtn").onclick	= function() { return false;};
-	YAHOO.util.Event.addListener("applyFormatBtn", "click", this.saveFilters.validateAndSaveFilters, this.saveFilters, this.saveFilters);
-	
-	// Fix z-index problem on Public Report Generator without changing css loading order.
-	this.fixZIndex("#new2_mask", 3);
+Filters.prototype.showSettings = function (reportContextId) {
+    initFormatPopup();
+    this.saveFilters = new SaveFilters(this, true);
+    var element = document.getElementById("customFormat");
+    element.style.display = "inline";
+    this.settingsPanel.setHeader(this.settingsPanelName);
+    this.settingsPanel.setBody(element);
+    this.settingsPanel.center();
+    this.settingsPanel.show();
+
+    YAHOO.util.Event.removeListener("applyFormatBtn", "click");
+    document.getElementById("applyFormatBtn").onclick = function () {

```

Comment:
Che if we can remove applyFormatBtn and the associated events

Reply:
Just confirmed is used for settings too.